src/Component/Router.purs

```diff
@@ -69,4 +68,4 @@ component =
       { route } <- H.get 
       when (route /= dest) do
```

Comment:
This is an example of what I'm talking about in preventing unnecessary renders / calls by guarding with the `when` function.